content/en/blog/_posts/2021-04-22-gateway-api/2021-04-22-gateway-api.md

```diff
@@ -0,0 +1,199 @@
+    - serviceName: foo-home
+      port: 8080
+  - matches:
+    - path:
+        type: Prefix
+        value: /
+    forwardTo:
+    - serviceName: foo-404
+      port: 8080
+```
+
+
+The bar team, operating in the bar Namespace of the same Kubernetes cluster, also wishes to expose their application to the internet, but they also want to control their own canary and blue-green rollouts. The following HTTPRoute is configured for the following behavior:
+
+*   For traffic to `bar.example.com`:
+    *   Send 90% of the traffic to bar-v1
+    *   Send 10% of the traffic to bar-v2
+*   For traffic to `bar.example.com` with the HTTP header `env: canary`:
+    
+    *   Send all the traffic to bar-v2
+    
+![httproute](./httproute.png)
+
+
+
+```
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: bar-route
+  namespace: bar
+  labels:
+    gateway: external-https-prod
+spec:
+  hostnames:
+  - "bar.example.com"
+  rules:
+  - forwardTo:
+    - serviceName: bar-v1
+      port: 8080
+      weight: 90
+    - serviceName: bar-v2
+      port: 8080
+      weight: 10
+  - matches:
+    - headers:
+        values:
+          env: canary
+    forwardTo:
+    - serviceName: bar-v2
+      port: 8080
+```
+
+
+
+### Route and Gateway Binding
+
+So we have two HTTPRoutes matching and routing traffic to different Services. You might be wondering, where are these Routes accessible? Through which networks or IPs are they exposed? 
+
+How Routes are exposed to clients is governed by [Route binding](https://gateway-api.sigs.k8s.io/concepts/api-overview/#route-binding), which describes how Routes and Gateways create a bidirectional relationship between each other. When Routes are bound to a Gateway it means their collective routing rules are configured on the underlying load balancers or proxies and the Routes are accessible through the Gateway. Thus, a Gateway is a logical representation of a networking data plane that can be configured through Routes.
+
+
+
+
+![route and gateway binding](./route-binding.png )
+
+### Administrative Delegation
+
+The split between Gateway and Route resources allows the cluster administrator to delegate some of the routing configuration to individual teams while still retaining centralized control. The following Gateway resource exposes HTTPS on port 443 and terminates all traffic on the port with a certificate controlled by the cluster administrator.
+
+
+```
+kind: Gateway
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  listeners:  
+  - protocol: HTTPS
+    port: 443
+    routes:
+      kind: HTTPRoute
+      selector:
+        matchLabels:
+          gateway: external-https-prod
+      namespaces:
+        from: All
+    tls:
+      certificateRef:
+        name: admin-controlled-cert
+```
+
+
+The following HTTPRoute shows how the Gateway selects the Route via it’s kind (HTTPRoute) and resource labels (gateway=external-https-prod). 
```

Comment:
```suggestion
The following HTTPRoute shows how the Gateway selects the Route via its `kind` (HTTPRoute) and resource labels (`gateway=external-https-prod`).
```

I'd say that the following manifest snippet is an example of an HTTPRoute that the Gateway's selector would match.

Reply:
How about:
```suggestion
The following HTTPRoute shows how the Route can ensure it matches the Gateway's selector via it’s `kind` (HTTPRoute) and resource labels (`gateway=external-https-prod`). 
```